mantidproject/mantid
# Update sanitizers to use compiler instrumentation

**Description of work.**
Updates the various sanitizers to use compiler instrumentation so we can re-enable the static analysis jobs on Jenkins

**To test:**
- You need to be running a *nix distribution with access to GCC-8 onwards (e.g. Ubuntu 18.04)
- Follow the instructions in the developer documentation, if anything is ambiguous or confusing please comment so I can clarify for future users
- Verify the right flags are being passed to the compiler when `USE_SANITIZER` is set by running
`ninja -nv | grep "fsanitize"` or `make -nv | grep "fsanitize"`
- Check the `RelWithDebInfo` in a sanitized mode switched to `-O1` and emits `-fno-omit-frame-pointer`

*There is no associated issue.*


*This does not require release notes* because this is an internal only change

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
